pkg/bitcoin/electrum/electrum.go

```diff
@@ -566,6 +566,38 @@ func (c *Connection) getScriptMempool(
 	return convertedItems, nil
 }
 
+// EstimateSatPerVByteFee returns the estimated sat/vbyte fee for a
+// transaction to be confirmed within the given number blocks.
+func (c *Connection) EstimateSatPerVByteFee(blocks uint32) (int64, error) {
+	// According to Electrum protocol docs, the returned fee is BTC/KB.
+	btcPerKbFee, err := requestWithRetry(
+		c,
+		func(
+			ctx context.Context,
+			client *electrum.Client,
+		) (float32, error) {
+			return client.GetFee(ctx, blocks)
+		})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get fee: [%v]", err)
+	}
+
+	// According to Electrum protocol docs, if the daemon does not have
+	// enough information to make an estimate, the integer -1 is returned.
+	if btcPerKbFee < 0 {
+		return 0, fmt.Errorf(
+			"daemon does not have enough information to make an estimate",
+		)
+	}
+
+	// To convert from BTC/KB to sat/vbyte, we need to multiply by 1e8/1e3.
+	satPerVByte := (1e8 / 1e3) * float64(btcPerKbFee)
+	// Make sure the minimum returned sat/vbyte fee is always 1.
+	satPerVByte = math.Max(satPerVByte, 1)
```

Comment:
This is dynamic and I don't think there is a range that makes sense. I checked the 3y fee history on testnet and it is very broad (source https://mempool.space/testnet/graphs/mining/block-fee-rates#3y):
<img width="1598" alt="image" src="https://github.com/keep-network/keep-core/assets/11180469/2b364299-5a83-4b17-b146-0fc154ce33fd">

I don't think that checking if the value is in the range `[1, 30000] ` tests this conversion in any way. On the other hand, if we use a narrower range, we may observe random failures of the integration test. In my opinion, the only thing that makes sense is asserting the value is `>=1` and that's it.

Reply:
What about this:
1. Let's add an integration test proving this mechanism works on all Electrum implementations ([this comment](https://github.com/keep-network/keep-core/pull/3581#discussion_r1201592389)).
2. Let's add a unit test that covers this logic: https://github.com/keep-network/keep-core/blob/43043cbbd06fa7387d118d2aa08361d452502b70/pkg/bitcoin/electrum/electrum.go#L585-L598
  Since the `Connection struct` refers to `client *electrum.Client` we can't easily inject the mock there to test the entire interaction. We can, however, introduce a package-private `convertBtcKBToSatVByte` function (rename if you can think of any better name) that will contain this logic and be tested from `electrum_test.go`. Not a perfect solution but better than nothing, especially since this logic is very important.  `convertBtcKBToSatVByte` would be called from `EstimateSatPerVByteFee` on the result returned by the Electrum server.

Reply:
See c61ec3b7